### ISSUE TITLE
Conditionally disable `maybe-with-title`

### ISSUE THREAD

**Issue by hkupty** [metosin/spec-tools#198]
Hi, first of all thanks for spec-tools.

Currently, we have a setup where we generate json-schema/swagger documentation from spec the following way:
- Every spec we declare that contains a `{:name ...}` attribute becomes a top-level schema;
- When a top-level schema is used to compose another top-level schema such as `(s/keys :req-un [::something])`, it is declared as a `$ref` instead of inlining the definition.

This works fine for most specs. However, those that we declare as `s/keys` or `s/map-of` end up including a stringified name of itself when it doesn't have an explicit `:name` assigned.

To circumvent that, I've created this local version:
```clj
(defn transform-no-inferred-title [schema]
  (with-redefs [spec-tools.json-schema/maybe-with-title (fn [schm _] schm)]
   (jsc/transform schema)))
```

However, I think it'd be best if it was something like this:
```
(binding [spec-tools.json-schema/*infer-titles* false]
  (jsc/transform schema))
```

or 

```
(jsc/transform schema {:infer-titles false})
```

Does this make sense?

Thanks in advance,
Henry Kupty

**Comment by ikitommi** [comment]
Explicit option would be great. PR welcome.

**Comment by miikka** [comment]
Fixed in 0.10.2 by adding a new parameter to `jsc/transform`:

```clojure
(jsc/transform schema {:infer-titles false})
```
